python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -203,7 +203,7 @@ def _formatted_multipartitioned_path(partition_key: MultiPartitionKey) -> str:
 
         asset_path = self._get_path_without_extension(context)
         return {
-            partition: self._with_extension(
+            partition.replace("/", "|"): self._with_extension(
```

Comment:
Thoughts on instead refactoring `formatted_partition_keys` above to map the partition key to the formatted path?

Then, here we can just do 
```
{
    partition: self._with_extension(
        self.get_path_for_partition(context, asset_path, partition_path)
    )
    for partition, partition_path in formatted_partition_keys.items()
}
```
so we can isolate all multipartition specific logic to `_formatted_multipartitioned_path`

Reply:
Thank you for your comment. Let me know if the changes look okay